### ISSUE TITLE
Compiler should report base method calls in ref struct declarations

### ISSUE THREAD

**Issue by mazhou-msft** [dotnet/roslyn#27049]
**Version Used**: 
Visual Studio 2017 d15.7+

**Steps to Reproduce**:

1. Create a C# project
2. Add a `ref struct` declaration, override any of the base methods and inside the body, call base method. for example:
```C#
public readonly ref struct S
{
    public override bool Equals(object obj) => base.Equals(obj);

    public override int GetHashCode() => base.GetHashCode();

    public override string ToString() => base.ToString();
}
```
3. Call any of these methods in a context, for example:
```C#
public static void Main(string[] args)
{
    S s = default;
    s.GetHashCode();
}
```

**Expected Behavior**:
Compiler should report the usage of the base method invocations inside the `ref struct`, issue a warning or an error to disallow the invocation of base methods from `System.Object`.

**Actual Behavior**:
You will get a runtime error: `System.InvalidProgramException`: Cannot create boxed by-ref like object.

**Comment by jcouv** [comment]
@OmarTawfik Would you be able to look at one more issue for 15.8? Thanks

**Comment by OmarTawfik** [comment]
@jcouv I'll take this one.
